src/EditorFeatures/Core/InheritanceMargin/InheritanceMarginServiceHelpers.cs

```diff
@@ -0,0 +1,338 @@
+    {
+        public static async ValueTask<ImmutableArray<SerializableInheritanceMarginItem>> GetInheritanceMemberItemAsync(
+            Solution solution,
+            ProjectId projectId,
+            ImmutableArray<(SymbolKey symbolKey, int lineNumber)> symbolKeyAndLineNumbers,
+            CancellationToken cancellationToken)
+        {
+            var remoteClient = await RemoteHostClient.TryGetClientAsync(solution.Workspace, cancellationToken).ConfigureAwait(false);
+            if (remoteClient != null)
+            {
+                var result = await remoteClient.TryInvokeAsync<IRemoteInheritanceMarginService, ImmutableArray<SerializableInheritanceMarginItem>>(
+                    solution,
+                    (remoteInheritanceMarginService, solutionInfo, cancellationToken) =>
+                        remoteInheritanceMarginService.GetInheritanceMarginItemsAsync(solutionInfo, projectId, symbolKeyAndLineNumbers, cancellationToken),
+                    cancellationToken).ConfigureAwait(false);
+
+                if (!result.HasValue)
+                {
+                    return ImmutableArray<SerializableInheritanceMarginItem>.Empty;
+                }
+
+                return result.Value;
+            }
+            else
+            {
+                return await GetInheritanceMemberItemInProcAsync(solution, projectId, symbolKeyAndLineNumbers, cancellationToken).ConfigureAwait(false);
+            }
+        }
+
+        private static async ValueTask<ImmutableArray<SerializableInheritanceMarginItem>> GetInheritanceMemberItemInProcAsync(
+            Solution solution,
+            ProjectId projectId,
+            ImmutableArray<(SymbolKey symbolKey, int lineNumber)> symbolKeyAndLineNumbers,
+            CancellationToken cancellationToken)
+        {
+            var project = solution.GetRequiredProject(projectId);
+            var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
+            using var _ = ArrayBuilder<SerializableInheritanceMarginItem>.GetInstance(out var builder);
+            foreach (var (symbolKey, lineNumber) in symbolKeyAndLineNumbers)
+            {
+                var symbol = symbolKey.Resolve(compilation, cancellationToken: cancellationToken).Symbol;
+                if (symbol is INamedTypeSymbol namedTypeSymbol)
+                {
+                    await AddInheritanceMemberItemsForNamedTypeAsync(solution, namedTypeSymbol, lineNumber, builder, cancellationToken).ConfigureAwait(false);
+                }
+
+                if (symbol is IEventSymbol or IPropertySymbol or IMethodSymbol)
+                {
+                    await AddInheritanceMemberItemsForTypeMembersAsync(solution, symbol, lineNumber, builder, cancellationToken).ConfigureAwait(false);
+                }
+            }
+
+            return builder.ToImmutable();
+        }
+
+        private static async ValueTask AddInheritanceMemberItemsForNamedTypeAsync(
```

Comment:
i presume this code is effectively teh same, and is just a move, right?

Reply:
There are two small changes below in this function.
```
var baseSymbols = allBaseSymbols
                .WhereAsArray(symbol => !symbol.IsErrorType() && symbol.SpecialType is not (SpecialType.System_Object or SpecialType.System_ValueType or SpecialType.System_Enum));
```
Here I added an extra condition to filter the error type out.
Because right now if I have
``` class Bar : ISomethingNotTyped { } ```
It would show an margin with an error type. 

Another change is, 
``` var derivedSymbols = allDerivedSymbols.WhereAsArray(symbol => symbol.Locations.Any(l => l.IsInSource));```
This is added because during my daily use I saw these when navigating to IEnumerable, which I feel is not great.
![image](https://user-images.githubusercontent.com/24360909/115288288-2191a180-a106-11eb-9ee9-b24207f51950.png)